server/resp/src/main/java/org/infinispan/server/resp/commands/list/LMOVE.java

```diff
@@ -0,0 +1,133 @@
+                                                                List<byte[]> arguments) {
+
+      return lmoveAndReturn(handler, ctx, arguments, false);
+   }
+
+   protected CompletionStage<RespRequestHandler> lmoveAndReturn(Resp3Handler handler,
+                                                                ChannelHandlerContext ctx,
+                                                                List<byte[]> arguments,
+                                                                boolean rightLeft) {
+      byte[] source = arguments.get(0);
+      byte[] destination = arguments.get(1);
+
+      boolean txCache = handler.cache().getCacheConfiguration().transaction().transactionMode().isTransactional();
+      boolean sameList = Arrays.equals(source, destination);
+
+      if (!sameList && !txCache) {
+         // different lists and not tx cache, warn
+         Log.SERVER.lmoveConsistencyMessage();
+      }
+
+      final boolean isSourceLeft;
+      final boolean isDestinationLeft;
+
+      if (rightLeft) {
+         isSourceLeft = false;
+         isDestinationLeft = true;
+      } else {
+         // parse and validate RIGHT and LEFT arguments
+         final String sourceWhereFrom = new String(arguments.get(2)).toUpperCase();
+         final String destinationWhereFrom = new String(arguments.get(3)).toUpperCase();
+         isSourceLeft = LEFT.equals(sourceWhereFrom);
+         isDestinationLeft = LEFT.equals(destinationWhereFrom);
+         if ((!isSourceLeft && !RIGHT.equals(sourceWhereFrom)) || (!isDestinationLeft && !RIGHT.equals(
+               destinationWhereFrom))) {
+            RespErrorUtil.syntaxError(handler.allocator());
+            return handler.myStage();
+         }
+      }
+
+      EmbeddedMultimapListCache<byte[], byte[]> listMultimap = handler.getListMultimap();
+      if (sameList) {
+         CompletionStage<byte[]> performedCall;
+         if (isSourceLeft && isDestinationLeft) {
+            // peek first element and do nothing else
+            performedCall = listMultimap.index(source, 0);
+         } else if (!isSourceLeft && !isDestinationLeft) {
+            // peek last element and do nothing else
+            performedCall = listMultimap.index(source, -1);
+         } else {
+            // rotate from left (head->tail) to right or from right to left (tail->left)
+            performedCall = listMultimap.rotate(source, isSourceLeft);
+         }
+         return handler.stageToReturn(performedCall, ctx, Consumers.GET_BICONSUMER);
+      }
+
+      CompletionStage<Collection<byte[]>> pollCall;
+      if (isSourceLeft) {
+         pollCall = listMultimap.pollFirst(source, 1);
+      } else {
+         pollCall = listMultimap.pollLast(source, 1);
+      }
+
+      return pollCall.thenCompose(
+            pollResult -> {
+               if (pollResult == null) {
+                  handler.stageToReturn(CompletableFutures.completedNull(), ctx, Consumers.GET_BICONSUMER);
```

Comment:
Seems it's missing the return here.

Reply:
I don't understand, tests pass correctly

Reply:
Yeah, I think because we write the bytes to the socket saying the result is null. But proceeding with the execution throws an NPE, goes to `RespHandler#exceptionCaught`, and closes the client connection. This will force the client to reconnect to keep operating.

Reply:
sorry I was unable to see the missing return before, you are right, just a forgotten return